h2o-core/src/main/java/water/util/ModelPngWriter.java

```diff
@@ -0,0 +1,81 @@
+package water.util;
+
+import hex.Model;
+import water.api.StreamWriter;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ModelPngWriter implements StreamWriter {
+
+    private final Model model;
+
+    public ModelPngWriter(Model model) {
+        this.model = model;
+    }
+
+    @Override
+    public void writeTo(OutputStream os) {
+        try {
+            Path pngtemp = Files.createTempFile("", "visualization.png");
+            model.visualize(pngtemp.toAbsolutePath().toString());
+
+            try (InputStream is = new FileInputStream(writeToZip(pngtemp).toFile())) {
+                byte[] buf = new byte[8192];
+                int c;
+                while ((c = is.read(buf, 0, buf.length)) > 0) {
+                    os.write(buf, 0, c);
+                    os.flush();
+                }
+                os.close();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
add an error message saying what went wrong - was it reading the png and writing it to stream?